advancedlog/main_logger.go

```diff
@@ -0,0 +1,122 @@
+package logger
+
+import (
+	"fmt"
+
+	logutils "github.com/bitrise-io/go-utils/v2/log"
+)
+
+// MainLogger ...
+type MainLogger struct {
+	internalLogger  SimplifiedLogger
+	debugLogEnabled bool
+}
+
+// NewMainLogger ...
+func NewMainLogger() MainLogger {
+	return MainLogger{
+		internalLogger:  newLegacyLogger(logutils.NewLogger()),
+		debugLogEnabled: false,
+	}
+}
+
+// Debug ...
+func (m *MainLogger) Debug(args ...interface{}) {
+	m.internalLogger.LogMessage(CLI, DebugLevel, fmt.Sprint(args...))
+}
+
+// Debugf ...
+func (m *MainLogger) Debugf(format string, args ...interface{}) {
+	m.internalLogger.LogMessage(CLI, DebugLevel, fmt.Sprintf(format, args...))
+}
+
+// Info ...
+func (m *MainLogger) Info(args ...interface{}) {
+	m.internalLogger.LogMessage(CLI, InfoLevel, fmt.Sprint(args...))
+}
+
+// Infof ...
+func (m *MainLogger) Infof(format string, args ...interface{}) {
+	m.internalLogger.LogMessage(CLI, InfoLevel, fmt.Sprintf(format, args...))
+}
+
+// Done ...
+func (m *MainLogger) Done(args ...interface{}) {
+	m.internalLogger.LogMessage(CLI, DoneLevel, fmt.Sprint(args...))
+}
+
+// Donef ...
+func (m *MainLogger) Donef(format string, args ...interface{}) {
+	m.internalLogger.LogMessage(CLI, DoneLevel, fmt.Sprintf(format, args...))
+}
+
+// Warn ...
+func (m *MainLogger) Warn(args ...interface{}) {
+	m.internalLogger.LogMessage(CLI, WarnLevel, fmt.Sprint(args...))
+}
+
+// Warnf ...
+func (m *MainLogger) Warnf(format string, args ...interface{}) {
+	m.internalLogger.LogMessage(CLI, WarnLevel, fmt.Sprintf(format, args...))
+}
+
+// Error ...
+func (m *MainLogger) Error(args ...interface{}) {
+	m.internalLogger.LogMessage(CLI, ErrorLevel, fmt.Sprint(args...))
+}
+
+// Errorf ...
+func (m *MainLogger) Errorf(format string, args ...interface{}) {
+	m.internalLogger.LogMessage(CLI, ErrorLevel, fmt.Sprintf(format, args...))
+}
+
+// Fatal ...
+func (m *MainLogger) Fatal(args ...interface{}) {
+	m.Error(args...)
+}
+
+// Fatalf ...
+func (m *MainLogger) Fatalf(format string, args ...interface{}) {
+	m.Errorf(format, args...)
+}
+
+// Print ...
+func (m *MainLogger) Print(args ...interface{}) {
+	m.internalLogger.LogMessage(CLI, NormalLevel, fmt.Sprintln(args...))
+}
+
+// Printf ...
+func (m *MainLogger) Printf(format string, args ...interface{}) {
+	m.internalLogger.LogMessage(CLI, NormalLevel, fmt.Sprintf(format, args...))
+}
+
+// Println ...
+func (m *MainLogger) Println(args ...interface{}) {
+	m.internalLogger.LogMessage(CLI, NormalLevel, fmt.Sprintln(args...))
+}
+
+// TPrintf ...
+func (m *MainLogger) TPrintf(format string, args ...interface{}) {
```

Comment:
Do we really need log functions with `T` prefix? Earlier it added time prefix to messages.

Reply:
I actually forgot to implement the time prefix which I will add because in the legacy log it will be still useful. For the new json logging I am hoping that we could add a time prefix to every line based on the time value in the json field. That would make the logs way more useful.

Reply:
Actually there is only one usage in the cli to log the timestamp. I will remove this function.